pkg/core/port.go

```diff
@@ -420,6 +420,23 @@ func (p *Port) assertChannelSpace() {
 	}
 }
 
+func (p *Port) AsyncWalkPrimitivePorts(handle func(p *Port)) {
+	if p.Primitive() {
+		go handle(p)
+	}
+
+	if p.Stream() != nil {
+		p.Stream().AsyncWalkPrimitivePorts(handle)
```

Comment:
@kairichard is `WalkPrimitivePorts` not sufficient enough? I guess `Async` is more a implementation detail

Reply:
@tidify IMO `walk/map/each` sound a lot like they should return something which in this case it does not. Maybe `Async` is also wrong, but at least it tripped me and might trip others as well.